### PR TITLE
Add inverse public websocket support

### PR DESCRIPTION
**Pull request by EricCrosson**

### CODE REVIEW

**Comment by EricCrosson** [comment]
Hey there :wave: 

Neat project! I added support for the public inverse websocket; it's functioning if not complete or perfect

**Comment by jukanntenn** [comment]
Nice job! I will do a review and verification as soon as possible.

File: src/inverse/ws.rs [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            if let Ok(ping) = rx.try_recv() {
+                ws.write_message(Message::Text(ping)).unwrap();
+            }
+
+            if let Ok(msg) = ws.read_message() {
+                match msg {
+                    Message::Text(content) => {
+                        if let Ok(_) = serde_json::from_str::<Pong>(&content) {
+                            continue;
+                        }
```
Comment (EricCrosson) [inline src/inverse/ws.rs:317]: I don't love this but it did clear a bunch of ERROR logs, for the cased where `Pong` messages weren't matched by `PublicResponse`.

Too bad line 325 doesn't match these messages :thinking:

Comment (jukanntenn) [reply src/inverse/ws.rs:317]: It's not a big problem, let's just make it work first and then take performance and code quality into account.

When I was verifying this PR, I found that instrument info snap response could not be deserialized and the response of perpetual is different from futures. I fixed them locally. I will merge this PR and push the patch to main branch.

Comment (EricCrosson) [reply src/inverse/ws.rs:317]: > let's just make it work first and then take performance and code quality into account

:rocket: 

> I fixed them locally. I will merge this PR and push the patch to main branch.

Thanks!

**Pull request merged**